www/include/views/componentTemplates/DB-Func.php

```diff
@@ -39,54 +39,42 @@
 
 function NameHsrTestExistence($name = null)
 {
-    global $pearDB, $pearDBO, $form;
-    $gsvs = null;
+    global $pearDB, $form;
+    $formValues = [];
+    $bindParams = [];
     if (isset($form)) {
-        $gsvs = $form->getSubmitValues();
+        $formValues = $form->getSubmitValues();
     }
-    $sql = "SELECT compo_id FROM giv_components_template WHERE ";
-    $sql .= "name = '" . $gsvs["name"] . "' ";
-    if ($gsvs["host_id"] != null) {
-        list($host_id, $service_id) = explode('-', $gsvs["host_id"]);
-        $sql .= "AND host_id = '" . $host_id . "' AND service_id = '" . $service_id . "'";
-    } else {
-        $sql .= "AND host_id IS NULL  AND service_id IS NULL";
-    }
-    $dbResult = $pearDB->query($sql);
-    $compo = $dbResult->fetch();
-    #Modif case
-    if ($dbResult->rowCount() >= 1 && $compo["compo_id"] == $gsvs["compo_id"]) {
-        return true;
-        #Duplicate entry
-    } elseif ($dbResult->rowCount() >= 1 && $compo["compo_id"] != $gsvs["compo_id"]) {
-        return false;
-    } else {
-        return true;
+    $query = 'SELECT compo_id FROM giv_components_template WHERE name = :name';
+    if (!empty($formValues['host_id'])) {
+        if (preg_match('/([0-9]+)-([0-9]+)/', $formValues['host_id'], $matches)) {
+            $formValues['host_id'] = (int) $matches[1];
+            $formValues['service_id'] = (int) $matches[2];
+        } else {
+            throw new \InvalidArgumentException('chartId must be a combination of integers');
+        }
     }
-}
 
-function DsHsrTestExistence($name = null)
-{
-    global $pearDB, $pearDBO, $form;
-    $gsvs = null;
-    if (isset($form)) {
-        $gsvs = $form->getSubmitValues();
+    $query .= (empty($formValues['host_id']) && empty($formValues['service_id']))
+        ? ' AND host_id IS NULL  AND service_id IS NULL'
+        : ' AND host_id = :host_id AND service_id = :service_id';
```

Comment:
If one of those 2 variables is empty you will go to your else `' AND host_id = :host_id AND service_id = :service_id'` and possibly bind an empty value